src/api/middleware/validators/validatorUtils.js

```diff
@@ -61,10 +61,35 @@ const isObjectId = (id) => {
  */
 const concurrentOffersNotExceeded = (OfferModel) => async (owner, publishDate, publishEndDate) => {
     // We need to pass the offer model in case we're inside an Offer instance
-    const concurrentOffers = await (new CompanyService())
+    const offersInTimePeriod = await (new CompanyService())
         .getOffersInTimePeriod(owner, publishDate, publishEndDate, OfferModel);
 
-    return concurrentOffers.length < CompanyConstants.offers.max_concurrent;
+    const offerNumber = offersInTimePeriod.length;
+    if (offerNumber < CompanyConstants.offers.max_concurrent) return true;
+    // otherwise, let's check how many are concurrent
+
+    // This algorithm is explained in https://github.com/NIAEFEUP/nijobs-be/issues/123#issuecomment-782272539
+
+    const offersByStart = offersInTimePeriod; // we won't need this array unmodified
+    const offersByEnd = [...offersInTimePeriod];
+    offersByStart.sort((offer1, offer2) => Date.parse(offer1.publishDate) - Date.parse(offer2.publishDate));
+    offersByEnd.sort((offer1, offer2) => Date.parse(offer1.publishEndDate) - Date.parse(offer2.publishEndDate));
```

Comment:
Create a `sortOffersByDateAscending` function before this function and just pass it to the .sort() to avoid duplicated code

Reply:
Although similar, the functions aren't exactly the same (since we use the publishDate in one and publishEndDate in the other). Is it worth it making a function with a custom property as an argument?

Reply:
Yep! Call it `sortOffersByFieldAscending` and you can use the `obj[arg]` notation to access the value of a varying arg in the object